### ISSUE TITLE
vanilla-server style .conf file completely ignored

### ISSUE THREAD

**Issue by Chais** [NyxStudios/TShock#1137]
I'm maintaining the AUR (Arch User Repository) package for tshock and tried to migrate to using vanilla-server style .conf files for compatibility. But I had to notice that any settings I made in a `.conf` file were completely ignored. Given a `.conf` file
```
port=7777
worldpath=/var/lib/tshock/Terraria/Worlds/
worldname=World1
autocreate=2
```
and starting the server with `mono TerrariaServer.exe -config /path/to/file.conf` I was presented with the interactive interface and the server created the directory `/var/lib/tshock/My Games/Terraria/Worlds` instead. No autocreation and the wrong worldpath.

**Comment by nicatronTg** [comment]
We support our own config file and command line arguments -- not the vanilla config file, if I recall correctly. This is because the API starts in a different way that allows us to override settings as we need.

Correct me if I'm wrong @NyxStudios/core

**Comment by WhiteXZ** [comment]
We support serverconfig.txt (the file that's generated by the Windows vanilla server) across all platforms via `TerrariaServer.exe -config /path/to/serverconfig.txt`. Try creating your config as `.txt` instead of `.conf`

**Comment by Chais** [comment]
Nope. Same issue. Furthermore I'd appreciate if vanilla `.conf` files were used because it allows for easy migration between vanilla and TShock server. Just point it to the `.conf`

**Comment by WhiteXZ** [comment]
OS: Ubuntu 14.04
Mono: 4.2.1
Using the following file, named serverconfig.txt
```
world=/home/white/My Games/Terraria/Worlds/trap-test.wld
maxplayers=8
port=7777
```
mono TerrariaServer.exe -config serverconfig.txt
Config was loaded.
I believe the issue lies in the parameters being specified. IIRC there's a very specific set you need in order for autocreation to work (and I'm working on figuring out what they are)

For the record, serverconfig using `.txt` is vanilla on Windows. We don't do any special checks to determine if you're running Windows or Unix-based, so using `.conf` as a default would cause issues with Windows users

**Comment by WhiteXZ** [comment]
Looks like autocreation from a config file is currently broken

**Comment by Chais** [comment]
Just like autocreation form cammand line #995 #1009

**Comment by WhiteXZ** [comment]
Command line autocreate worked fine for me, though I did not examine the world. I'll look at fixing config autogen after I finish my exams

**Comment by ProfessorXZ** [comment]
This issue was fixed in [7776c6f](https://github.com/NyxStudios/TerrariaAPI-Server/commit/7776c6fa4703de32a20cf981a6cd18d7527a4340)

**Comment by Chais** [comment]
Can't confirm.
```
$ tshock -configpath /var/lib/tshock -config /etc/conf.d/tshock/default.conf
TerrariaAPI Version: 1.24.0.0 (Protocol v1.3.3.2 (176))
Terraria Server v1.3.3.2


n           	New World
d   <number>	Delete World

Choose World: 
```
default.conf:
```
port=7777
world=/var/lib/tshock/worlds/World1.wld
worldpath=/var/lib/tshock/Terraria/Worlds/
worldname=World1
autocreate=2
```
